### ISSUE TITLE
change the peers.json servlet: read from settings/peers.json instead of RemoteAccess.history

### ISSUE THREAD

**Issue by Orbiter** [loklak/loklak_server#471]
with https://github.com/loklak/loklak_server/issues/470 the peers information becomes persistent. That will enable us to contact peers which had been seen in the past.
To use that information, we must do:
- if a peer calls api/hello.json, we make an entry in settings/peers.json using the access methods from  https://github.com/loklak/loklak_server/issues/470
- if the api/peers.json is called, we return information from settings/peers.json

**Comment by treba123** [comment]
Before we can write a hash to the peers.json, we need to make some kind of handshake. Because the hash is only of benefit if we make sure the peer has the corresponding private key.

A handshake would be something like:
- peer call api/hello.json and posts it's public key
- the server sends a random string back
- the peer encrypts the string with it private key and sends it back to api/hello.json
- the server decrypts the string with the public key and validates it
- if everything went well, the server can write the public key and the corresponding hash to settings/peers.json

see https://github.com/loklak/loklak_server/issues/456

**Comment by Orbiter** [comment]
That may block the process which retrieves the key. Instead, just write a 'validated' property so the key can be validated asynchronously

**Comment by Orbiter** [comment]
With validated=false if not yet validated. Wie can then run the process as you suggested in the cleanup loop.

**Comment by Orbiter** [comment]
Let me give an example: consider I get a peer list from a (trusted/untrusted) source. That could be a list of 200 peers. To check the validity of all peers a background process should be considered. Such a background process would need a work-list: wich peers to be checked. So I believe we need such a process anyway. That would make it useful to write down any new peer to the list with an 'untrusted' flag.

**Comment by treba123** [comment]
Ah alright, the peer list is public and we want be able ask a server for its known peers.

Then we have two different usecases here:
1. A new peer Alice calles Bobs api/hello.json: In this case I'd suggest to do a handshake if Alice is not known already before writing the data to settings/peers.json (for ddos reasons, see https://github.com/loklak/loklak_server/issues/456). We only need to do that once, when a peer connects to a server for the first time, afterwards we can retrieve the data.
2. Alice calls Bobs api/peers.json: Alice wants to know about all peers known by Bob. In this case, we can maybe even just trust Bob for the time being, and check the keys later if we need to.

For usecase 2. I'd suggest two more things:
- Bob should put a signature on the list. So we know that this is indeed Bobs list of known hosts. That's something we can basicly do with every api that is only used retrieve data. But probably has time.
- If api/peers.json is a public api, we should probably only include peer hashes/public keys. The ip is something not everybody needs to know, especially if we want peers to be able to contribute anonymously. Otherwise it will get incredible easy to get a full list of ip-addresses of the whole network.

**Comment by Orbiter** [comment]
This is sound but adds too mich complexity to the simple requirement initially requested.
That means: too much efford right now.
We must do authentication first.
Lets postpone this.

**Comment by Orbiter** [comment]
To be reopened if there is time to do more p2p.

**Comment by Orbiter** [comment]
Turned this into an epic: add single issues with reference to this epic later
